### PR TITLE
Adding the Issue Metrics Action to generate metrics on how we manage issues (Lombiq Technologies: OCORE-139)

### PR DESCRIPTION
**Pull request by Piedone**
Fixes https://github.com/OrchardCMS/OrchardCore/issues/14585

Here's an example issue that this generates: https://github.com/OrchardCMS/OrchardCore/issues/15073 Alternatively, or additionally, this can be added to a job summary: https://github.com/OrchardCMS/OrchardCore/actions/runs/7495302237 This alone might not be enough, because then we can't discuss it, and it's less visible.

### CODE REVIEW

File: .github/workflows/issue-metrics.yml [hunk 1]
```diff
@@ -0,0 +1,61 @@
+name: Monthly Issue and Pull Request Metrics
+on:
+  pull_request:
+    branches: [ main, release/** ]
```
Comment (Piedone) [inline .github/workflows/issue-metrics.yml:4]: This is only for testing, needs to be removed before merge.

**Comment by Piedone** [comment]
This needs to be fixed before we can adopt it: https://github.com/github/issue-metrics/issues/188. Or we can just accept that the "Time to answer" metric is broken for now, though I think that's quite crucial.

Here's are example issues:

- https://github.com/OrchardCMS/OrchardCore/issues/15051
- https://github.com/OrchardCMS/OrchardCore/issues/15052

**Comment by Piedone** [comment]
Misunderstood it, we have no bug: https://github.com/github/issue-metrics/issues/188#issuecomment-1885936756.

**Comment by Piedone** [comment]
@agriffard can you review this? It would be great to have it merged still in January, because then we'll see if it works as the scheduled workflow on the 1st February.

File: .github/workflows/issue-metrics.yml [hunk 2]
```diff
@@ -0,0 +1,81 @@
...
+        Add-Content -Path ./community_metrics.md -Value (Get-Content -Path ./issue_metrics.md -Raw)
+
+    - name: Compute Pull Request Metrics
+      uses: github/issue-metrics@v2
+      env:
+        GH_TOKEN: ${{ secrets.GITHUB_TOKEN }}
+        SEARCH_QUERY: 'repo:OrchardCMS/OrchardCore is:pr created:${{ env.LAST_MONTH }} -label:dontmerge -label:notready -is:draft'
+        HIDE_TIME_TO_ANSWER: true
+
+    - name: Concatentate Issue and Pull Request Metrics
```
Comment (hishamco) [inline .github/workflows/issue-metrics.yml:50]:
```suggestion
    - name: Concatenate Issue and Pull Request Metrics
```

File: .github/workflows/issue-metrics.yml [hunk 3]
```diff
@@ -0,0 +1,81 @@
...
+        Add-Content -Path ./community_metrics.md -Value ([Environment]::NewLine + $content)
+
+    - name: Compute Q&A Discussion Request Metrics
+      uses: github/issue-metrics@v2
+      env:
+        GH_TOKEN: ${{ secrets.GITHUB_TOKEN }}
+        SEARCH_QUERY: 'repo:OrchardCMS/OrchardCore type:discussions created:${{ env.LAST_MONTH }} category:Q&A'
+        HIDE_TIME_TO_CLOSE: true
+
+    - name: Concatentate Issue/PR and Discussion Metrics
```
Comment (hishamco) [inline .github/workflows/issue-metrics.yml:63]:
```suggestion
    - name: Concatenate Issue/PR and Discussion Metrics
```

**Review by hishamco — CHANGES_REQUESTED** [review]

**Review by hishamco — APPROVED** [review]

**Comment by Piedone** [comment]
Thanks!

**Pull request merged**